src/main/kotlin/org/rust/cargo/runconfig/RsExecutableRunner.kt

```diff
@@ -74,9 +78,20 @@ abstract class RsExecutableRunner(
 
         val runCargoCommand = state.prepareCommandLine()
         val (_, executableArguments) = parseArgs(runCargoCommand.command, runCargoCommand.additionalArguments)
+
+        val workingDirectory = if (runCargoCommand.command == "test") {
+            val srcPath = artifact?.target?.src_path
+            val file = srcPath?.let { LocalFileSystem.getInstance().refreshAndFindFileByPath(it) }
+            val pkg = file?.let { environment.project.cargoProjects.findPackageForFile(it) }
+            val rootDirectory = pkg?.rootDirectory
+            rootDirectory ?: runCargoCommand.workingDirectory
+        } else {
+            runCargoCommand.workingDirectory
+        }
```

Comment:
nit: Let's extract into separate method

Reply:
I don't insist but just note that getting a proper working directory is still no extracted to separate method

Reply:
Oh, I forgot about it. Fixed.